hera_sim/simulate.py

```diff
@@ -231,14 +252,19 @@ def __init__(
                     self.data.data_array[:] = 0.0
                     self.data.flag_array[:] = False
                     self.data.nsample_array[:] = 1.0
-            elif self.data is not None:
+            else:
                 self.data = data
 
         # Assume the phase type is drift unless otherwise specified.
         if self.data.phase_type == "unknown":
             self.data.set_drift()
 
+        # what does this line do?
```

Comment:
I have a feeling it might be doing some initialization, but I can't remember for sure. Can you test with it deleted? If nothing bad happens, remove it, if so, comment it!

Reply:
nothing bad happened on removing it, so I removed it for good.